src/libraries/System.Text.Json/gen/JsonSourceGenerator.Emitter.cs

```diff
@@ -107,24 +110,20 @@ public void Emit()
                 foreach (ContextGenerationSpec contextGenerationSpec in _generationSpec.ContextGenerationSpecList)
                 {
                     _currentContext = contextGenerationSpec;
-
-                    bool generateGetConverterMethodForTypes = false;
-                    bool generateGetConverterMethodForProperties = false;
+                    _generateGetConverterMethodForTypes = false;
+                    _generateGetConverterMethodForProperties = false;
 
                     foreach (TypeGenerationSpec typeGenerationSpec in _currentContext.RootSerializableTypes)
                     {
                         GenerateTypeInfo(typeGenerationSpec);
-
-                        generateGetConverterMethodForTypes |= typeGenerationSpec.HasTypeFactoryConverter;
-                        generateGetConverterMethodForProperties |= typeGenerationSpec.HasPropertyFactoryConverters;
```

Comment:
Yeah quite subtle as this only captured types specified with `[JsonSerializable]`.